covariancematrix.go

```diff
@@ -12,15 +12,17 @@ import (
 	"github.com/gonum/matrix/mat64"
 )
 
-// CovarianceMatrix calculates a covariance matrix (also known as a
-// variance-covariance matrix) from a matrix of data, using a two-pass
-// algorithm.
+// CovarianceMatrix returns the covariance matrix (also known as the
+// variance-covariance matrix) calculated from a matrix of data, x, using
+// a two-pass algorithm.
 //
-// The weights must have length equal to the number of rows in
-// input data matrix x. If cov is nil, then a new matrix with appropriate size will
-// be constructed. If cov is not nil, it should have the same number of columns as the
-// input data matrix x, and it will be used as the destination for the covariance
-// data. Weights must not be negative.
+// If weights is not nil the weighted covariance of x is calculated. weights
+// must have length equal to the number of rows in input data matrix and
+// must not contain negative elements.
+// If cov is zero sized, a new matrix with appropriate size will be constructed.
```

Comment:
Sure, but it's also not obvious that `nil` is zero-sized. The operations in mat64 don't allow nil arguments, so I don't think it's obvious that this does. Maybe "If cov is nil or zero sized, a new ..."